### PR TITLE
Week 3 Fragments

### PR DESCRIPTION
**Pull request by JoshGoldsmith**
frau

### CODE REVIEW

**Review by pelusodan — DISMISSED** [review]

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -9,6 +9,9 @@
         android:roundIcon="@mipmap/ic_launcher_round"
         android:supportsRtl="true"
         android:theme="@style/Theme.PlayboiKotlin">
+        <activity
+            android:name=".josh.fragments.JoshFragmentsActivity"
+            android:exported="false" />
```
Comment (robert-chatterton) [inline app/src/main/AndroidManifest.xml:14]: josh

File: app/src/main/java/jayp/bge/playboikotlin/josh/JoshActivity.kt [hunk 2]
```diff
@@ -17,25 +17,37 @@ class JoshActivity : AppCompatActivity() {
...
-        binding.newButton.setOnClickListener {
-            viewModel.onNButtonPressed()
+    private fun observeViewState(viewState: JoshViewState) {
+        binding.textView.text =
+            "I have clicked " + viewState.clicks + " time and the current name is " + viewState.name
+        if (viewState.isLoading) {
+            binding.progRockBar.isVisible = true
+            binding.progRockBar.animate()
+        } else {
+            binding.progRockBar.isVisible = false
```
Comment (robert-chatterton) [inline app/src/main/java/jayp/bge/playboikotlin/josh/JoshActivity.kt:34]: prog rock

File: app/src/main/java/jayp/bge/playboikotlin/josh/JoshActivity.kt [hunk 3]
```diff
@@ -17,25 +17,37 @@ class JoshActivity : AppCompatActivity() {
...
+        with(binding) {
+            joshButton.setOnClickListener {
+                viewModel.onJButtonPressed()
+            }
+            newButton.setOnClickListener {
+                viewModel.onNButtonPressed()
+            }
+            goToFragmentsButton.setOnClickListener {
+                val intent = Intent(this@JoshActivity, JoshFragmentsActivity::class.java)
+                startActivity(intent)
```
Comment (robert-chatterton) [inline app/src/main/java/jayp/bge/playboikotlin/josh/JoshActivity.kt:50]: button

File: app/src/main/java/jayp/bge/playboikotlin/josh/fragments/http/HttpFragment.kt [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        savedInstanceState: Bundle?
+    ): View {
+        binding = FragmentHttpBinding.inflate(layoutInflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        binding.joshButton.setOnClickListener {
```
Comment (robert-chatterton) [inline app/src/main/java/jayp/bge/playboikotlin/josh/fragments/http/HttpFragment.kt:28]: button

File: app/src/main/java/jayp/bge/playboikotlin/josh/fragments/https/HttpsFragment.kt [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+class HttpsFragment : Fragment() {
+
+    private lateinit var binding: FragmentHttpsBinding
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        // Inflate the layout for this fragment
+        binding = FragmentHttpsBinding.inflate(layoutInflater, container, false)
```
Comment (robert-chatterton) [inline app/src/main/java/jayp/bge/playboikotlin/josh/fragments/https/HttpsFragment.kt:19]: bind

File: app/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -13,7 +13,9 @@
     <string name="title_web_loader">Web Loader</string>
     <string name="title_list_o_rama">List-o-rama</string>
     <string name="title_chatty">Chatty</string>
-    <!-- TODO: Remove or change this placeholder text -->
+    <string name="title_http">HTTP</string>
+    <string name="title_https">HTTPS</string>
+    <string name="empty_text">Enter Text Here</string>
```
Comment (robert-chatterton) [inline app/src/main/res/values/strings.xml:18]: way too much xml

**Review by robert-chatterton — APPROVED** [review]
josh

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (JoshGoldsmith) [reply app/src/main/AndroidManifest.xml:14]: i don't believe you

File: app/src/main/java/jayp/bge/playboikotlin/josh/JoshActivity.kt (hunk 2 above)
Comment (JoshGoldsmith) [reply app/src/main/java/jayp/bge/playboikotlin/josh/JoshActivity.kt:34]: i love yes

File: app/src/main/java/jayp/bge/playboikotlin/josh/JoshActivity.kt (hunk 3 above)
Comment (JoshGoldsmith) [reply app/src/main/java/jayp/bge/playboikotlin/josh/JoshActivity.kt:50]: nottub

File: app/src/main/java/jayp/bge/playboikotlin/josh/fragments/http/HttpFragment.kt (hunk 4 above)
Comment (JoshGoldsmith) [reply app/src/main/java/jayp/bge/playboikotlin/josh/fragments/http/HttpFragment.kt:28]: nottub

File: app/src/main/java/jayp/bge/playboikotlin/josh/fragments/https/HttpsFragment.kt (hunk 5 above)
Comment (JoshGoldsmith) [reply app/src/main/java/jayp/bge/playboikotlin/josh/fragments/https/HttpsFragment.kt:19]: ing

File: app/src/main/res/values/strings.xml (hunk 6 above)
Comment (JoshGoldsmith) [reply app/src/main/res/values/strings.xml:18]: i disagree

**Pull request merged**